utils/api/requests.js

```diff
@@ -72,3 +72,100 @@ export const sendMessage = ({ id, message, files }) => {
 
   posting(`/quote_groups/${id}/notes.json`, note)
 }
+
+export const useInitializeRequest = (id) => {
+  const { data, error } = useSWR(`/wares/${id}/quote_groups.json`, fetcher)
+  let dynamicForm = { name: data?.name }
+
+  if (data?.dynamic_form) {
+    const defaultSchema = data.dynamic_form.schema
+    const defaultOptions = data.dynamic_form.options
+    const schema = dynamicFormSchema(defaultSchema)
+
+    dynamicForm = {
+      ...dynamicForm,
+      schema,
+      uiSchema: dynamicFormUiSchema(schema, defaultOptions),
+    }
+  }
+
+  return {
+    dynamicForm,
+    isLoadingInitialRequest: !error && !dynamicForm,
+    isInitialRequestError: error,
+  }
+}
+
+// TODO(alishaevn): https://github.com/assaydepot...scientist_api_v2/app/serializers/scientist_api_v2/dynamic_form_serializer.rb#L39
+// update the method at the code above to return the configured schema below
+export const dynamicFormSchema = (defaultSchema) => {
+  // TODO(alishaevn): may need to account for multiple forms
+
+  const removedProperties = [
+    'concierge_support', 'suppliers_identified', 'price_comparison', 'number_suppliers',
+    'supplier_criteria', 'supplier_confirmation'
+  ]
+  let propertyFields = {}
+  let requiredFields = []
+  let dependencyFields = {}
+
+  Object.entries(defaultSchema.properties).forEach(prop => {
+    const [key, value] = prop
+    let adjustedProperty
+
+    if (!removedProperties.includes(key)) {
+      if (value.required) {
+        requiredFields.push(key)
+        let { required, ...remainingProperties } = value
+        adjustedProperty = { ...remainingProperties }
```

Comment:
I started off using `delete value[key]`. but for some reason that kept resulting in `requiredFields = []` and `dependencyFields = {}` on the form. (see second row in image below)
![Screen Shot 2023-01-12 at 12 32 07 PM](https://user-images.githubusercontent.com/29032869/212178008-ea33a0a7-50eb-4e44-8ab9-d85fdabb2735.png)